cmd/check/main.go

```diff
@@ -49,29 +49,53 @@ func main() {
 		}
 	}
 
-	ref, err := name.ParseReference(req.Source.Name(), name.WeakValidation)
+	repo, err := name.NewRepository(req.Source.Repository, name.WeakValidation)
 	if err != nil {
-		logrus.Errorf("could not resolve repository/tag reference: %s", err)
+		logrus.Errorf("failed to resolve repository: %s", err)
 		os.Exit(1)
 		return
 	}
 
+	var origin *name.Registry
+	if req.Source.RegistryMirror != "" {
+		registry, err := name.NewRegistry(req.Source.RegistryMirror, name.WeakValidation)
+		if err != nil {
+			logrus.Errorf("could not resolve registry: %s", err)
+			os.Exit(1)
+			return
+		}
+
+		origin = new(name.Registry)
+		*origin = repo.Registry
+		repo.Registry = registry
+	}
+
+	ref := repo.Tag(req.Source.Tag())
+
 	var response CheckResponse
-	err = resource.RetryOnRateLimit(func() error {
+	op := func() error {
```

Comment:
I missed this earlier, but I am thinking its not desirable for the function to be a closure but rather have its arguments called out in the signature.

Reply:
Are you suggesting that the `op` variable be a function with 1 parameter (`ref`)? Or something else?

Since there's a similar strategy used in `in`, I'm assuming the same change would apply there, too.